done: registration

ym-guotao opened the pull request.

ym-guotao committed:
done: registration

ym-guotao committed:
fix test bug

ym-guotao committed:
add prettier settings

qixiuying commented on src/registerServiceWorker.js:
```diff
@@ -0,0 +1,108 @@
+// In production, we register a service worker to serve assets from local cache.
```
what this used for?

ym-guotao replied:
This model was added by create-react-app.
In production, we register a service worker to serve assets from local cache.
This lets the app load faster on subsequent visits in production, and gives it offline capabilities. However, it also means that developers (and users) will only see deployed updates on the "N+1" visit to a page, since previously cached resources are updated in the background.
To learn more about the benefits of this model,  read https://goo.gl/KwvDNy.

qixiuying commented on src/App.js:
```diff
@@ -0,0 +1,169 @@
+              must be numbers and length equal to 11.
+            </FormText>
+          </FormGroup>
+          <Button 
```
from the wiki, the button should be disabled first, only all conditions passed, enable it

ym-guotao replied:
create a variable named btnDisable to listen whether or not the form validation passed

qixiuying commented on src/index.test.js:
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
```
from wiki, need write test with jest

ym-guotao replied:
sorry a little confuse, the project is using Jest to do test. Do you mean we need write more test case?

qixiuying replied:
I mean no any jest functions you used in the test. Your current test doesn't have any valid check. I think should at least use `https://facebook.github.io/jest/docs/en/snapshot-testing.html#content` to check whether the result is same as expected.

And better test the valid check functions and submit button click function.  
Suggest to use jest + enzyme to simulate the actions and check the result, it is easier to add tests.

And better comment the coverage in pr comments. Then the reviewer is more clear for the test coverage.

ym-guotao replied:
test submit function by dispatching action  store.dispatch(accountFormSubmit(some example JSON))

ym-guotao committed:
bugfix: disable the submit button until the form validation passed.

qixiuying commented on src/App.js:
```diff
@@ -1,169 +1,163 @@
+              id="userName"
               placeholder="please input your name"
-              onInput={(event) => this.onFieldChange('userName', event.target.value)}
+              onInput={event =>
```
just suggest, use onBlur for the event trigger, and better show error message if it does not satisfy the requirements.

ym-guotao replied:
improved according to suggestion

ym-guotao committed:
add test cases.  
coverage => Lines: 66.67%.
 coverage => Funcs: 47.06%

ym-guotao committed:
update readme

qixiuying reviewed (approved):
👍

ym-guotao committed:
add yarn

ym-guotao committed:
add flow

ym-guotao committed:
eject create react app, hence can do CSS Modules

ym-guotao committed:
CSS modules bug fix

ym-guotao committed:
fix test bug after using CSS modules.
And for the devDependencies issue, I just found this:
https://github.com/facebookincubator/create-react-app/issues/2696
the conclusion is CRA remove devDependencies on purpose.
